packages/iris-grid/src/IrisGridUtils.ts

```diff
@@ -288,7 +312,26 @@ class IrisGridUtils {
   static hydrateIrisGridState(
     model: IrisGridModel,
     irisGridState: DehydratedIrisGridState
-  ): Partial<IrisGridProps> {
+  ): Pick<
+    IrisGridState,
+    | 'advancedFilters'
+    | 'customColumnFormatMap'
+    | 'isFilterBarShown'
+    | 'quickFilters'
+    | 'sorts'
+    | 'customColumns'
+    | 'conditionalFormats'
+    | 'reverseType'
+    | 'rollupConfig'
+    | 'showSearchBar'
+    | 'searchValue'
+    | 'selectDistinctColumns'
+    | 'selectedSearchColumns'
+    | 'invertSearchColumns'
+    | 'pendingDataMap'
+    | 'frozenColumns'
+    | 'aggregationSettings'
+  > & { userColumnWidths: ModelSizeMap; userRowHeights: ModelSizeMap } {
```

Comment:
Can make this a bit cleaner - I think the only discrepancy is around `metrics` and the `userColumnWidths` and `userRowHeights`:
```suggestion
  ): Omit<HydratedIrisGridState, 'metrics'> & {
    userColumnWidths: ModelSizeMap;
    userRowHeights: ModelSizeMap;
  } {
```